WordPress/Classes/ViewRelated/Reader/ReaderMenuViewModel.swift

```diff
@@ -0,0 +1,615 @@
+        fetchRequest.predicate = predicateForRequests()
+        let sortDescriptor = NSSortDescriptor(key: "title", ascending: true, selector: #selector(NSString.localizedCaseInsensitiveCompare))
+        fetchRequest.sortDescriptors = [sortDescriptor]
+
+        tagsFetchedResultsController = NSFetchedResultsController(fetchRequest: fetchRequest,
+                                                                  managedObjectContext: ContextManager.sharedInstance().mainContext,
+                                                                  sectionNameKeyPath: nil,
+                                                                  cacheName: nil)
+        tagsFetchedResultsController.delegate = self
+
+        do {
+            let _ = try tagsFetchedResultsController.performFetch()
+        } catch {
+            DDLogSwift.logError("There was a problem fetching tag topics for the menu.")
+            assertionFailure("There was a problem fetching tag topics.")
+        }
+    }
+
+
+    /// Updates the lists results controller's fetch request predicate and performs a new fetch
+    ///
+    func updateAndPerformTagsFetchRequest() {
+        tagsFetchedResultsController.fetchRequest.predicate = predicateForRequests()
+        do {
+            let _ = try tagsFetchedResultsController.performFetch()
+        } catch {
+            DDLogSwift.logError("There was a problem fetching tag topics for the menu.")
+            assertionFailure("There was a problem fetching tag topics.")
+        }
+    }
+
+
+    /// Sets up the tags section and its corresponding NSFetchedResultsController.
+    ///
+    func setupTagsSection() {
+        let section = ReaderMenuSection(title: NSLocalizedString("Tags", comment:"Section title of the tags reader section."), type: .Tags)
+        sections.append(section)
+    }
+
+
+    /// Returns the number of items for the tags section.
+    ///
+    /// - Returns: The number of items in the section.
+    ///
+    func itemCountForTagSection() -> Int {
+        return tagsFetchedResultsController.fetchedObjects?.count ?? 0
+    }
+
+
+    /// Returns the menu item from the tags section at the specified index.
+    ///
+    /// - Parameters:
+    ///     - index: The index of the item.
+    ///
+    /// - Returns: The requested menu item or nil.
+    ///
+    func menuItemForTagAtIndex(index: Int) -> ReaderMenuItem? {
+        guard let topic = tagsFetchedResultsController.objectAtIndexPath(NSIndexPath(forRow: index, inSection: 0)) as? ReaderAbstractTopic else {
+            return nil
+        }
+        return ReaderMenuItem(title: topic.title, type: .Topic, icon: nil, topic: topic)
+    }
+
+
+    // MARK: - Helper Methods
+
+
+    /// Check if the user is signed in
+    ///
+    func isSignedInWPCom() -> Bool {
+        let service = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        return service.defaultWordPressComAccount() != nil
```

Comment:
Funny thing... I realized after I made the PR that I had an almost duplicate method in ReaderHelpers. 
The facepalm is real.